hack/install-wrapper.ps1

```diff
@@ -0,0 +1,194 @@
+        $script:Cloud = $AWS
+    } elseif ($vendor -like "*Microsoft*Corporation") {
+        $script:Cloud = $AZURE
+    } else {
+        Log "Unknown cloud platform. Only AWS and Azure Clouds are supported"
+        ExitOnError
+    }
+}
+
+function UpdateAntreaURL()
+{
+    $branchTag = $AntreaVersion.Substring(1,3)
+    $Script:AntreaBranch = "release-${branchTag}"
+    $Script:AntreaInstallScript = "https://raw.githubusercontent.com/antrea-io/antrea/${AntreaBranch}/hack/externalnode/install-vm.ps1"
+    $Script:AntreaConfig = "https://raw.githubusercontent.com/antrea-io/antrea/${AntreaBranch}/build/yamls/externalnode/conf/antrea-agent.conf"
+    $Script:AgentBin = "https://github.com/antrea-io/antrea/releases/download/${AntreaVersion}/antrea-agent-windows-x86_64.exe"
+}
+
+function GenerateNodename()
+{
+    $name = $null
+    $retries = 5
+    if ($Cloud -eq $AWS) {
+        for ($i=0; $i -lt $retries; $i++) {
+            try {
+                $endPoint = "$AWSMetaRoot/instance-id"
+                # NodeName is represented as vm-<instance-id>
+                $name = "vm-" + (Invoke-WebRequest -uri $endPoint).Content
+                break
+            } catch {
+                Log "Retrying query for $endPoint, attempt=$i"
+                Start-Sleep -Seconds 5
+                continue
+            }
+        }
+    } elseif ($Cloud -eq $AZURE) {
+        $vmId=""
+        $vmName=""
+        for ($i=0; $i -lt $retries; $i++) {
+            try {
+                $endPoint = "$AzureMetaRoot/instance/compute/resourceId?api-version=2021-01-01&format=text"
+                $vmId = (Invoke-RestMethod -Headers @{"Metadata"="true"} -Method GET -Uri $endPoint).ToLower()
+                $endPoint = "$AzureMetaRoot/instance/compute/name?api-version=2021-01-01&format=text"
+                $vmName = (Invoke-RestMethod -Headers @{"Metadata"="true"} -Method GET -Uri $endPoint).ToLower()
+                break
+            } catch {
+                Log "Retrying query for $endPoint, attempt=$i"
+                Start-Sleep -Seconds 5
+                continue
+            }
+        }
+        # Convert to ASCII
+        $vmIdAscii = [int[]][char[]]$vmId
+        # Compute sum of ASCII values of vmId
+        $sum=0
+        foreach ($i in $vmIdAscii) {
+            $sum = $sum + $i
+        }
+        # NodeName is represented as vm-<vm-name>-<hash of the VM resource ID>
+        $name = "vm-" + $vmName + "-" + $sum
+    }
+    $Script:Nodename = $name
+    if ($Nodename -eq "") {
+        Log "ERROR! Nodename cannot be empty for cloud $Cloud"
+        ExitOnError
+    }
+}
+
+function DownloadFile($src, $dst, $options)
+{
+    Log "Downloading file $src to $dst"
+    curl.exe $options "$src" -o "$dst"
+}
+
+function DownloadAntreaFiles()
+{
+    RemoveIfExists $AntreaTemp -recurse $true
+    New-Item $AntreaTemp -type directory -Force | Out-Null
+    DownloadFile $AntreaInstallScript $AntreaTemp/$InstallScript
```

Comment:
The options are required only to download the binary files, since github uses redirections for large files. 
While for the others are normal files, redirection option is not needed.

Reply:
Kept the curl options common for all the files that are downloaded.